File: src/main/java/com/example/storage/controller/EmployeeController.java
package com.example.storage.controller;


import com.example.storage.controller.dto.EmployeeDto;
import com.example.storage.service.EmployeeService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("api")
@RequiredArgsConstructor
public class EmployeeController {

    private final EmployeeService employeeService;

    @GetMapping("/{code}/auth")
    @ResponseStatus(code = HttpStatus.OK)
    public void login(@PathVariable String code) {
        employeeService.auth(code);
    }

    @GetMapping("/{code}/info")
    @ResponseStatus(code = HttpStatus.OK)
    public EmployeeDto getByCode(@PathVariable String code) {
        return employeeService.getByCode(code);
    }

}

File: src/main/kotlin/ru/nto/storage/controller/EmployeeController.kt
package ru.nto.storage.controller

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController
import ru.nto.storage.controller.dto.EmployeeDto
import ru.nto.storage.service.EmployeeService

@RestController
@RequestMapping("/api")
class EmployeeController(
    private val employeeService: EmployeeService
) {

    @GetMapping("/{code}/auth")
    @ResponseStatus(HttpStatus.OK)
    fun login(@PathVariable code: String) {
        employeeService.auth(code)
    }

    @GetMapping("/{code}/info")
    @ResponseStatus(HttpStatus.OK)
    fun getByCode(@PathVariable code: String): EmployeeDto {
        return employeeService.getByCode(code)
    }
}
